jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/LinkPredictionAlgorithm.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (C) Copyright 2020-2020, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg.interfaces;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jgrapht.alg.util.Pair;
+import org.jgrapht.alg.util.Triple;
+
+/**
+ * A link prediction algorithm.
+ * 
+ * <p>
+ * A link prediction algorithm assigns a score $s_{uv}$ for each pair of nodes $u, v \in V$. This
```

Comment:
This definition could be improved a bit.  

1) The interface as defined assigns a score for a given pair or for a list of pairs (not for each possible pair in V).  

2) The references to existence vs non-existence are confusing.  Is it correct to state that the edges present in the graph are observed links, and the metric returned indicates the relative scoring for a given unobserved link being a true link?

And I don't see where the actual sorting happens?  Nor do I see a test for the list of pairs case.